test/e2e/config/azure-dev.yaml

```diff
@@ -165,10 +163,6 @@ providers:
         targetName: "cluster-template-conformance-presubmit-artifacts.yaml"
       - sourcePath: "${PWD}/templates/test/dev/cluster-template-custom-builds.yaml"
         targetName: "cluster-template-conformance-presubmit-artifacts-windows-containerd.yaml"
```

Comment:
not related to this change but @jsturtevant do you know why we're writing the `cluster-template-custom-builds.yaml` template above twice with two different names?

Reply:
Prior to this we needed different templates for windows and so we tacked windows to the end: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/30bfbb55d43f10343c634a3f90396ff206444021/test/e2e/conformance_test.go#L125-L127

After this merged we might not need this logic anymore as we can re-use the same templates.  I will create an issue to follow up on this after this merges

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2315